src/routes/users/fetch-users.e2e.test.js

```diff
@@ -0,0 +1,59 @@
+const { expect } = require('chai')
+const { fixtures, setupDB } = require('../../../test/utils')
+const fetchUsers = require('../../services/users/fetch-users')
+
+describe('GET /users', () => {    
+	describe('Success', () => {
+		let adminUser
+		let accessToken
+		beforeEach(async () =>{
+			await setupDB()
+			await fixtures.insertUser()
+			adminUser = await fixtures.insertUser({
+				role : 'admin'
+			}) 
+			const body = { id: adminUser.id, email: adminUser.email }
+			accessToken = fixtures.generateAccessToken(
+				{user : body}
+			)
+		})
+		it('should return 200 if users are fetched', async () => 
+			fixtures.api()
+				.get(`/api/v1/users`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.expect(200)
+		)
+		it('should return the right number of users', async() =>{
+			const actualUsers = await fetchUsers()
+			fixtures.api()
+				.get(`/api/v1/users`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.then(res => {
+				 expect(res.body.length).eql(actualUsers.length)
+				}) 
+		})	
+	})
+	describe('Failure', ()=> {
+		let userUser
+		let accessToken
+		beforeEach(async () =>{
```

Comment:
This works but I would say let's try to evaluate cases where a `before` would suffice, since it's not always we might need to clear the database and insert again for some tests

Reply:
i am not sure what you mean

Reply:
by using a `beforeEach` here, you mean a new user should be inserted before each of the test cases which might not be necessary based on the tests you have

alternatively, you could have the user inserted once by using a `before` block which runs once before all the test cases you have. It helps to reduce the amount of work being done and can help improve test speed

it depends on the tests though, hence me saying we should not just default to `beforeEach` without there being a good use case for it

Reply:
oh okay, i get you now. Thanks so much for the explanation